File: October/Week2/Day1/binarySearch.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        
        def binarySearch(nums, l, h, k):
            # print(nums,l,h,k)
            if h >= l:
                mid = l + (h - l) // 2
                
                if nums[mid] == k:
                    return mid
                elif nums[mid] > k:
                    return binarySearch(nums,l,mid-1,target)
                elif nums[mid] < k:
                    return binarySearch(nums,mid+1,h,target)
            return -1
            
        low = 0
        high = len(nums)
        # print(binarySearch(nums,low,high,target))
        return binarySearch(nums,low,high-1,target)

File: October/Week2/Day1/binarySearch.java
class Solution {
    public int binarySearch(int[] nums, int l, int h, int target) {
        if(h >= l) {
            int mid = (l + h) / 2;
            
            if(nums[mid] == target) {
                return mid;
            }
            
            if (nums[mid] > target) {
                return binarySearch(nums, l, mid-1, target);
            }
            
            if (nums[mid] < target) {
                return binarySearch(nums, mid+1, h, target);
            }
        }
        
        return -1;
    }
    public int search(int[] nums, int target) {
        int low = 0;
        int high = nums.length;
        
        return binarySearch(nums,low,high-1,target);
    }
}